### PR TITLE
Feature/108891 status manager class

### PR DESCRIPTION
**Pull request by jesseeproductions**
-108891
-108892

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Each line in an array declaration must end in a comma | [Manager.php:138](https://github.com/moderntribe/event-tickets/blob/e8348d58de3107b2d4eeef6b45eda89213e3365b/src/Tribe/Status/Manager.php#L138)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Status/Manager.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		'WooCommerce' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Status_Manager',
+	);
+
+
+	/**
+	 * An array of status objects for all active commerces
+	 *
+	 * @var array
+	 */
+	protected $statues = array();
```
Comment (borkweb) [inline src/Tribe/Status/Manager.php]: I think this is meant to be `$statuses`?

File: src/Tribe/Status/Manager.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	/**
+	 * Gets the ticket provider class when passed an id
+	 *
+	 * @since TBD
+	 *
+	 * @param integer|string $id a rsvp order key, order id, attendee id, ticket id, or product id
+	 *
+	 * @return bool|object
+	 */
+	protected function get_provider_by_id( $id ) {
```
Comment (borkweb) [inline src/Tribe/Status/Manager.php]: This method seems like an odd fit in this class.

File: src/Tribe/Status/Manager.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		return tribe( 'tickets.data_api' )->get_ticket_provider( $id );
+	}
+
+	public function get_trigger_statuses( $commerce ) {
```
Comment (borkweb) [inline src/Tribe/Status/Manager.php]: Doc block?

File: src/Tribe/Status/Template.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+
+/**
+ * Class Tribe__Tickets__Status__Template
+ *
+ * @since TBD
+ *
+ */
+abstract class Tribe__Tickets__Status__Template {
```
Comment (borkweb) [inline src/Tribe/Status/Template.php]: "Template" carries a lot of baggage and using it here could cause confusion. How about `Tribe__Tickets__Status__Abstract`?

File: src/Tribe/Status/Manager.php [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	/**
+	 * An array of status objects for WooCommerce Tickets
+	 *
+	 * @var array
+	 */
+	protected $status_managers = array(
+		//'EDD' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Status_Manager',
+		//'RSVP' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Status_Manager',
+		//'Tribe Commerce' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Status_Manager',
+		'WooCommerce' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Status_Manager',
```
Comment (borkweb) [inline src/Tribe/Status/Manager.php]: I know I saw this before, but I just thought of it again. Perhaps we should use the commerce provider slugs that we have standardized within the ET suite?

* `woo`
* `edd`
* `rsvp`
* `tribe-commerce`

**Pull request closed without merging**